### DISCUSSION TITLE
initialState中的InitialStateType无法推导TS

### DISCUSSION THREAD

**Discussion by blackcorgi** [umijs/umi#12459] [Q&A]
getInitialState中通过类型声明也无法在useModel使用时生效。

![043031C6-3282-4639-A1D5-D65466F21159](https://github.com/umijs/umi/assets/7530811/b176f9cf-07ab-4b0b-b096-f12be5ce26c7)
![2A4F0295-A900-439C-91D3-1A1AE3E864F2](https://github.com/umijs/umi/assets/7530811/bf541dcc-62fc-4c5d-8989-1297f5a8be5a)

**Comment by fz6m** [comment]
一个新生成的 umi 4 max 项目里，有一段代码：

```ts
// app.ts

export async function getInitialState(): Promise<{ name: string }> {
  return { name: '@umijs/max' };
}
```

然后在其他地方使用 `useModel` 获取这个值的时候，是可以获取到类型提示的。

如果还有问题，需要给一个有类型问题的最小复现仓库。

**Reply by wooktao** [reply]
![image](https://github.com/user-attachments/assets/db7bc6b5-d202-4c78-890b-836abdabae23)
![image](https://github.com/user-attachments/assets/857d034f-331c-4065-b259-8cc35333cbcf)

**Reply by wooktao** [reply]
使用的就是默认生成的
